src/Generators/AbstractTagGenerator.php

```diff
@@ -176,21 +178,32 @@ public function getExistingTags()
     /**
      * Generate the Owner-properties for extensions.
      *
-     * @throws NotFoundExceptionInterface
+     * @throws ReflectionException
      */
     protected function generateOwnerTags()
     {
         $className = $this->className;
         // If className is abstract, Injector will fail to instantiate it
-        $reflection = new \ReflectionClass($className);
+        $reflection = new ReflectionClass($className);
         if ($reflection->isAbstract()) {
             return;
         }
         if (Injector::inst()->get($this->className) instanceof Extension) {
             $owners = array_filter(DataObjectAnnotator::getExtensionClasses(), function ($class) use ($className) {
-                $config = Config::inst()->get($class, 'extensions');
-
-                return ($config !== null && in_array($className, $config, null));
+                $config = Config::inst()->get(
+                    $class,
+                    'extensions',
+                    Config::UNINHERITED | Config::EXCLUDE_EXTRA_SOURCES
+                );
+                if (!$config) {
+                    return false;
+                }
+                foreach ($config as $candidateClass) {
+                    if (Extension::get_classname_without_arguments($candidateClass) === $className) {
+                        return true;
```

Comment:
Maybe, returning a return variable makes more sense? Also, this loop is unreliable. I feel a variable would make more sense, to keep the code semantically readable.

Reply:
I think that makes sense actually. Updating.

Reply:
Or I'll make it a generator instead of array_map.